api/src/main/java/org/openmrs/api/db/hibernate/HibernateFormDAO.java

```diff
@@ -434,8 +435,7 @@ private Criteria getFormCriteria(String partialName, Boolean published, Collecti
 	        Boolean retired, Collection<FormField> containingAnyFormField, Collection<FormField> containingAllFormFields,
 	        Collection<Field> fields) {
 		
-		Criteria crit = sessionFactory.getCurrentSession().createCriteria(Form.class, "form");
-		
+		Criteria crit = sessionFactory.getCurrentSession().createCriteria(Form.class, "f");
```

Comment:
What is the reason for changing from "form" to "f"?

Reply:
Basically the logic in the fix is this, The FormField Hibernate domain(Table) has a field <code>form</code>. Since the parent Criteria is owned by Form Hibernate domain/table, if we use an alliance of the same name as a field contained in the FormField domain(Detached  Criteria) Hibernate will get confused and do some thing like in the sql String below.

<code>select *  from form this_ where this_.published=? and ? = (select count(ff_.form_field_id) as y0_ from form_field ff_ where ff_.form_id = ff_.form_id and ff_.form_field_id in (?, ?, ?, ?)) </code>

instead of :
<code>select *  from form this_ where this_.published=? and ? = (select count(ff_.form_field_id) as y0_ from form_field ff_ where ff_.form_id = this_.form_id and ff_.form_field_id in (?, ?, ?, ?)) </code>

So just need another name/alliance apart from 'form'

Reply:
ping @dkayiwa